nltk/twitter/util.py

```diff
@@ -0,0 +1,362 @@
+    to a CSV file for easier processing. For example, just tweetIDs or
+    just the text content of the tweets can be extracted.
+
+    Additionally, the function allows combinations of fields of other Twitter
+    objects (mainly the users, see below).
+
+    For Twitter entities (e.g. hashtags of a tweet), and for geolocation, see
+    `json2csv_entities`
+
+    :param str infile: The name of the file containing full tweets
+
+    :param str outfile: The name of the text file where results should be\
+    written
+
+    :param list fields: The list of fields to be extracted. Useful examples\
+    are 'id_str' for the tweetID and 'text' for the text of the tweet. See\
+    <https://dev.twitter.com/overview/api/tweets> for a full list of fields.
+    e. g.: ['id_str'], ['id', 'text', 'favorite_count', 'retweet_count']
+    Addionally, it allows IDs from other Twitter objects, e. g.,\
+    ['id', 'text', {'user' : ['id', 'followers_count', 'friends_count']}]
+
+
+    :param error: Behaviour for encoding errors, see\
+    https://docs.python.org/3/library/codecs.html#codec-base-classes
+    """
+    (writer, outf) = outf_writer_compat(outfile, encoding, errors)
+    for line in fp:
+        tweet = json.loads(line)
+        row = extract_fields(tweet, fields)
+        writer.writerow(row)
+    outf.close()
+
+def outf_writer_compat(outfile, encoding, errors):
+    """
+    Identify appropriate CSV writer given the Python version
+    """
+    if compat.PY3 == True:
+        outf = open(outfile, 'w', encoding=encoding, errors=errors)
+        writer = csv.writer(outf)
+    else:
+        outf = open(outfile, 'wb')
+        writer = compat.UnicodeWriter(outf, encoding=encoding, errors=errors)
+    return (writer, outf)
+
+
+
+
+def json2csv_entities(fp, outfile, main_fields, entity_name, entity_fields,
+                      encoding='utf8', errors='replace'):
+    """
+    Extract selected fields from a file of line-separated JSON tweets and
+    write to a file in CSV format.
+
+    This utility function allows a file of full tweets to be easily converted
+    to a CSV file for easier processing of Twitter entities. For example, the
+    hashtags or media elements of a tweet can be extracted.
+
+    :param file-object fp: The name of the file containing full tweets
+
+    :param str outfile: The name of the text file where results should be\
+    written
+
+    :param list main_fields: The list of fields to be extracted from the main\
+    object, usually the tweet. Useful examples: 'id_str' for the tweetID. See\
+    <https://dev.twitter.com/overview/api/tweets> for a full list of fields.
+    e. g.: ['id_str'], ['id', 'text', 'favorite_count', 'retweet_count']
+    If `entity_name` is expressed as a dictionary, then it is list of fields\
+    of the object that corresponds to the key of the dictionary (could be\
+    the user object, or the place of a tweet object).
+
+    :param list entity_name: The name of the entity: 'hashtags', 'media',\
```

Comment:
Can we call this `entity_type`, rather than name? Presumably a mentioned user is the entity, and has a name.